eng/Publishing.props

```diff
@@ -1,7 +1,29 @@
 <?xml version="1.0" encoding="utf-8"?>
 <Project>
-   <PropertyGroup>
-      <PublishingVersion>3</PublishingVersion>
-      <ProducesDotNetReleaseShippingAssets>true</ProducesDotNetReleaseShippingAssets>
-   </PropertyGroup>
-</Project>
+  <PropertyGroup>
+    <PublishingVersion>3</PublishingVersion>
+    <ProducesDotNetReleaseShippingAssets>true</ProducesDotNetReleaseShippingAssets>
+    <PublishDependsOnTargets>$(PublishDependsOnTargets);_PublishPackages</PublishDependsOnTargets>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <_PackagesToPublish Remove="@(_PackagesToPublish)" />
+    <_PackagesToPublish Include="$(ArtifactsPackagesDir)**\*.nupkg" UploadPathSegment="Roslyn-analyzers" Condition="'$(DotNetBuildRepo)' == 'true'" />
```

Comment:
This is very similar to the `fsharp` case: https://github.com/dotnet/fsharp/pull/16838#discussion_r1518055210. Arcade infra won't collect all of the packages as the sub-directories under `artifacts/Packages` are different than what arcade expects.

Reply:
It shouldn't collect everything I think. This will include duplicates of packages (with different branding).

Reply:
Old model of harvesting artifacts (RepoManifest.xml) was collecting all of these. Here's are the packages:
```
artifacts/packages/Release/PreRelease/Microsoft.CodeAnalysis.Analyzers.3.11.0-beta1.final.nupkg
artifacts/packages/Release/PreRelease/Microsoft.CodeAnalysis.AnalyzerUtilities.3.11.0-beta1.final.nupkg
artifacts/packages/Release/PreRelease/Microsoft.CodeAnalysis.NetAnalyzers.9.0.0-preview.final.nupkg
artifacts/packages/Release/PreRelease/Roslyn.Diagnostics.Analyzers.3.11.0-beta1.final.nupkg
artifacts/packages/Release/Release/Microsoft.CodeAnalysis.Analyzers.3.11.0.nupkg
artifacts/packages/Release/Release/Microsoft.CodeAnalysis.AnalyzerUtilities.3.11.0.nupkg
artifacts/packages/Release/Release/Microsoft.CodeAnalysis.NetAnalyzers.9.0.0.nupkg
artifacts/packages/Release/Release/Roslyn.Diagnostics.Analyzers.3.11.0.nupkg
artifacts/packages/Release/Shipping/Microsoft.CodeAnalysis.Analyzers.3.11.0-beta1.24128.1.nupkg
artifacts/packages/Release/Shipping/Microsoft.CodeAnalysis.AnalyzerUtilities.3.11.0-beta1.24128.1.nupkg
artifacts/packages/Release/Shipping/Microsoft.CodeAnalysis.NetAnalyzers.9.0.0-preview.24128.1.nupkg
artifacts/packages/Release/Shipping/Roslyn.Diagnostics.Analyzers.3.11.0-beta1.24128.1.nupkg
```

No duplication, but I'm unsure if we need all of them. Linux VMR build did require more than what was harvested by default `arcade` publishing infra.

Reply:
OK - it appears that none of the other packages are needed from this repo - I'm going to close this PR once I'm done with verification.

Reply:
Tried to remove all of these packages and 2nd-stage Linux build failed with missing dependency (package downgrade):

`
##vso[task.logissue type=error;sourcepath=/src/git/dotnet/src/sdk/src/BuiltInTools/dotnet-watch/dotnet-watch.csproj;linenumber=0;columnnumber=0;code=NU1605;](NETCORE_ENGINEERING_TELEMETRY=Restore) Warning As Error: Detected package downgrade: Microsoft.CodeAnalysis.AnalyzerUtilities from 3.11.0 to 3.11.0-beta1.24128.1. Reference the package directly from the project to select a different version. %0A dotnet-watch -> Microsoft.CodeAnalysis.CSharp.Features 4.10.0-3.24129.7 -> Microsoft.CodeAnalysis.Features 4.10.0-3.24129.7 -> Microsoft.CodeAnalysis.AnalyzerUtilities (>= 3.11.0) %0A dotnet-watch -> Microsoft.CodeAnalysis.CSharp.Features 4.10.0-3.24129.7 -> Microsoft.CodeAnalysis.AnalyzerUtilities (>= 3.11.0-beta1.24128.1)
`

I am now testing a change to include the following packages, and only in source-only build:
```
artifacts/packages/Release/Release/Microsoft.CodeAnalysis.Analyzers.3.11.0.nupkg
artifacts/packages/Release/Release/Microsoft.CodeAnalysis.AnalyzerUtilities.3.11.0.nupkg
artifacts/packages/Release/Release/Microsoft.CodeAnalysis.NetAnalyzers.9.0.0.nupkg
artifacts/packages/Release/Release/Roslyn.Diagnostics.Analyzers.3.11.0.nupkg
```

Reply:
Trimmed the publishing to include minimum required packages - https://github.com/dotnet/roslyn-analyzers/pull/7233/commits/f62d4c4f329e65cf32a353296ad827ca7ca7fc6a

Without these changes, 2nd stage Linux build is failing. This can be evaluated at later time to determine if there is a way to get rid of these additional packages. Linux source-build was requiring all of these packages and more. This is an improvement.

Reply:
Added a reference to tracking issue with https://github.com/dotnet/roslyn-analyzers/pull/7233/commits/109356ee5515bb8885e80d331862da10fd8943d7